### PR TITLE
feat(search): search v2 with `@algolia/autocomplete-js`

### PR DESCRIPTION
**Pull request by charlypoly**

### CODE REVIEW

File: packages/components/src/components/SearchbarV2/SidePreview.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  components: AutocompleteComponents;
+  accentColor: string;
+}) => {
+  const project = PRODUCTS.find((p) => p.children === item.source);
+  return (
+    item && (
+      <PreviewContainer accentColor={accentColor}>
+        {project && (
+          <PreviewProject
+            key={project.children}
```
Comment (B2o5T) [inline packages/components/src/components/SearchbarV2/SidePreview.tsx:114]:
```suggestion
```
why use `key` in this place?

File: packages/components/src/components/SearchbarV2/SidePreview.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        </ContentSnippet>
+        {item.toc.length > 1 && (
+          <>
+            <TocTitle>Table of content:</TocTitle>
+            <TOC toc={item.toc} />
+          </>
+        )}
+        <Cta
+          accentColor={accentColor}
+        >{`Press "Enter" to open ${item.type} →`}</Cta>
```
Comment (B2o5T) [inline packages/components/src/components/SearchbarV2/SidePreview.tsx:142]:
```suggestion
        >Press "Enter" to open {item.type} →</Cta>
```

File: packages/components/src/components/SearchbarV2/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  css`
+    padding: var(--aa-spacing-half);
+    max-height: var(--aa-detached-modal-max-height);
+  `,
+]);
+
+export const ResultsContainer = styled.div([
+  css`
+    min-width: 400px;
+  `,
```
Comment (B2o5T) [inline packages/components/src/components/SearchbarV2/styles.tsx:14]: tw`min-w-[400px]`,

Comment (B2o5T) [inline packages/components/src/components/SearchbarV2/styles.tsx:14]:
```suggestion
  tw`min-w-[400px]`,
```

**Comment by B2o5T** [comment]
fixes https://github.com/the-guild-org/the-guild-components/issues/361 https://github.com/the-guild-org/the-guild-components/issues/362

File: packages/components/src/components/SearchBarV2.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      },
+    });
+    search.current = s;
+
+    return () => {
+      s.destroy();
+    };
+  }, [accentColor, placeholder]);
+
+  // listen for CTRL+K
```
Comment (B2o5T) [inline packages/components/src/components/SearchBarV2.tsx:166]: I fixed it, now cmd+k not only open modal but closes also

Comment (charlypoly) [reply packages/components/src/components/SearchBarV2.tsx:166]: Thanks @B2o5T !!

**Review by dotansimha — APPROVED** [review]

**Pull request merged**